File: web/src/app/candy/model/candy.ts
import { Amount } from "../../common/model/amount";

export interface CandyDashboard {
  balanceUsd: Amount
  balance: Amount
  lastExpenses: CandyExpense[]
}

export interface CandyExpense {
  id: string
  date: string
  amountUsd: Amount
  amount: Amount
  comment: string
}

File: server/src/main/kotlin/com/evgenltd/financemanager/candy/record/candy.kt
package com.evgenltd.financemanager.candy.record

import com.evgenltd.financemanager.common.util.Amount
import java.time.LocalDate
import java.util.*

data class CandyDashboardRecord(
    val balance: Amount,
    val balanceUsd: Amount,
    val lastExpenses: List<CandyExpenseRecord>
)

data class CandyExpenseRecord(
    val id: UUID?,
    val date: LocalDate,
    val amount: Amount,
    val amountUsd: Amount?,
    val comment: String?
)